src/HealthGPS.Console/poco.h

```diff
@@ -1,55 +1,83 @@
 #pragma once
+#include "HealthGPS.Core/interval.h"
+
+#include <filesystem>
 #include <map>
 #include <optional>
 #include <string>
 #include <unordered_map>
 #include <vector>
 
+/**
+ * @brief Data structures containing model parameters and configuration options
+ *
+ * POCO stands for "plain old class object". These structs represent data structures
+ * which are contained in JSON-formatted configuration files.
+ */
 namespace host::poco {
+//! Information about a data file to be loaded
 struct FileInfo {
-    std::string name;
+    std::filesystem::path name;
     std::string format;
     std::string delimiter;
-    std::string encoding;
     std::map<std::string, std::string> columns;
+
+    auto operator<=>(const FileInfo &rhs) const = default;
```

Comment:
This is a new C++20 thing. If you add a default spaceship operator, then it automatically generates other comparison functions too (i.e. `operator==`, `operator<` etc.): https://en.cppreference.com/w/cpp/language/default_comparisons

Oddly you don't get a default `operator==` otherwise, which seems a little odd to me.